pychromecast/__init__.py

```diff
@@ -334,6 +364,68 @@ def join(self, timeout=None):
         """
         self.socket_client.join(timeout=timeout)
 
+    def start_dynamic_host_tracking(self):
+        """
+        Start dynamic host tracking, register for connection status events
+        """
+        self.socket_client.register_connection_listener(self)
+
+    def new_connection_status(self, status):
+        """
+        Callback for socket client connection status
+        """
+        if not status.status == CONNECTION_STATUS_DISCONNECTED:
+            return  # Not interesting
+
+        def discovery_timeout():
+            """
+            Internal scheduled timeout for discovery, 5 seconds
+            """
+            time.sleep(10)
```

Comment:
Instead of starting a thread that sleeps for 10 seconds, consider using a [Timer](https://docs.python.org/3.4/library/threading.html#timer-objects)